pages/api/maintenance/index.js

```diff
@@ -0,0 +1,19 @@
+const Sentry = require('@sentry/node');
+
+const { checkMaintenanceModeGateway, sentryParams } = require('../gateways');
+
+export default async function (req, res) {
+  Sentry.init(sentryParams);
+
+  try {
+    let result = await checkMaintenanceModeGateway(req.body);
+    res.status(200).json(result)
+
+  } catch (e) {
+    Sentry.captureException(e);
+    await Sentry.flush(2000);
+
+    let result = new Error('Error checking');
+    res.status(500).json(result)
```

Comment:
So will this throw an error if the maintenance mode is down?

Reply:
What about a maintenance mode for maintenance mode? =D

Reply:
It is somewhat unlikely that the maintenance endpoint would throw any errors, since it only reads from an environment variable. But if it doesn't work, it should be picked up on sentry.

Plus once maintenance mode is enabled, I am expecting that the developer will manually check the frontend.

Reply:
But what if they network between the frontend & main endpoint is interrupted?

Reply:
Why shouldn't the logic be "if maint mode is true, activate maint mode", "if we don't know what mode, or it's explicitly off, disable maint mode"

Reply:
We need to get in to the mindset of giving our users the best service i.e. the highest uptime

Reply:
You're totally correct. The logic that you described was what I intended to write. 

Returning the result when the request failed on line 17 still wouldn't result in the maintenance page appearing, but it would throw an error. I have made a few changes to address this.

So if there was a network issue, the maintenance page not appear.